File: LeetCode/0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    bool isVowel(char c) {
        return c == 'a' || c == 'A' || c == 'e' || c == 'E' || c == 'i' ||
               c == 'I' || c == 'o' || c == 'O' || c == 'u' || c == 'U';
    }
    string reverseVowels(string s) {
        int l = 0, r = s.length() - 1;
        while (l <= r) {
            bool lVowel = isVowel(s[l]), rVowel = isVowel(s[r]);
            if (!lVowel)
                l++;
            if (!rVowel)
                r--;
            if(lVowel && rVowel) swap(s[l++],s[r--]);
        }
        return s;
    }
};

File: LeetCode/0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        left = 0
        right = len(s) - 1
        result = list(s)
        s = s.lower()
        VOWELS = ['a', 'e', 'i', 'o', 'u']

        while left < right:

            left_is_vowel = s[left] in VOWELS
            right_is_vowel = s[right] in VOWELS

            if left_is_vowel and right_is_vowel:
                result[left], result[right] = result[right], result[left]
                left += 1
                right -= 1
            elif not left_is_vowel:
                left += 1
            else:
                right -= 1
        
        return "".join(result)
                